examples/chef/common/chef-rpc-actions-worker.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "chef-rpc-actions-worker.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/callback.h>
+#include <app/data-model/Nullable.h>
+#include <app/util/config.h>
+#include <lib/core/DataModelTypes.h>
+#include <map>
+#include <platform/CHIPDeviceLayer.h>
+
+using chip::app::DataModel::Nullable;
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::rpc;
+
+static std::map<ClusterId, ActionsDelegate *> gActionsDelegateMap{};
+
+ActionsDelegate * RpcFindActionsDelegate(ClusterId clusterId)
+{
+    if (gActionsDelegateMap.find(clusterId) != gActionsDelegateMap.end())
+    {
+        return gActionsDelegateMap[clusterId];
+    }
+
+    return nullptr;
+}
+
+static void RpcActionsTaskCallback(System::Layer * systemLayer, void * data)
+{
+    ChefRpcActionsWorker * worker = (ChefRpcActionsWorker *) data;
+
+    worker->ProcessActionQueue();
+}
+
+bool ChefRpcActionsCallback(EndpointId endpointId, ClusterId clusterId, uint8_t type, uint32_t delayMs, uint32_t actionId,
+                            std::vector<uint32_t> args)
+{
+    ActionTask task(endpointId, clusterId, static_cast<ActionType>(type), delayMs, actionId, args);
+
+    return ChefRpcActionsWorker::Instance().EnqueueAction(task);
+}
+
+bool ChefRpcActionsWorker::EnqueueAction(ActionTask task)
+{
+    bool kickTimer = false;
+
+    if (queue.empty())
+    {
+        kickTimer = true; // kick timer when the first task is adding to the queue
+    }
+
+    queue.push(task);
+
+    if (kickTimer)
+    {
+        (void) DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(task.delayMs), RpcActionsTaskCallback, this);
+    }
+    return true;
+}
+
+void ChefRpcActionsWorker::ProcessActionQueue()
+{
+    // Dequeue the first item
+    ActionTask task = queue.front();
+    queue.pop();
+
+    ActionsDelegate * delegate = RpcFindActionsDelegate(task.clusterId);
+    if (nullptr == delegate)
```

Comment:
Please do an early return on this `if` condition and remove (de-nest) the `else`.